modules/auxiliary/server/capture/smb.rb

```diff
@@ -25,626 +28,254 @@ def initialize
         authenticate. Another option is using auxiliary/spoof/{nbns,llmnr} to
         respond to queries for names the victim is already looking for.
+
+        Documentation of the above spoofing methods can be found by running `info -d`.
       },
-      'Author' => 'hdm',
+      'Author' => [
+        'hdm',                 # Author of original module
+        'zeroSteiner',         # Creator of RubySMB::Server
+        'agalway-r7',          # Port of existing module to use RUBYSMB::Server
+        'sjanusz-r7',          # Port of existing module to use RUBYSMB::Server
+      ],
       'License' => MSF_LICENSE,
-      'Actions' => [[ 'Capture', 'Description' => 'Run SMB capture server' ]],
+      'Actions' => [[ 'Capture', { 'Description' => 'Run SMB capture server' } ]],
       'PassiveActions' => [ 'Capture' ],
       'DefaultAction' => 'Capture'
     })
 
     register_options(
       [
-        OptString.new('CAINPWFILE',  [ false, "The local filename to store the hashes in Cain&Abel format", nil ]),
-        OptString.new('JOHNPWFILE',  [ false, "The prefix to the local filename to store the hashes in John format", nil ]),
-        OptString.new('CHALLENGE',   [ true, "The 8 byte server challenge", "1122334455667788" ])
-      ])
-
-    register_advanced_options(
-      [
-        OptBool.new("SMB_EXTENDED_SECURITY",
-          [ true,
-            "Use smb extended security negotiation, when set client will use " \
-            "ntlmssp, if not then client will use classic lanman " \
-            "authentification",
-            false
-          ]),
-        OptBool.new("NTLM_UseNTLM2_session",
-          [ true,
-            "Activate the 'negotiate NTLM2 key' flag in NTLM authentication. " \
-            "When SMB_EXTENDED_SECURITY negotiate is set, client will use " \
-            "ntlm2_session instead of ntlmv1 (default on win 2K and above)",
-            false
-          ]),
-        OptBool.new("USE_GSS_NEGOTIATION",
-          [ true,
-            "Send a gss_security blob in smb_negotiate response when SMB " \
-            "extended security is set. When this flag is not set, Windows will " \
-            "respond without gss encapsulation, Ubuntu will still use gss.",
-            true
-          ]),
-        OptString.new('DOMAIN_NAME',
-          [ true,
-            "The domain name used during smb exchange with SMB_EXTENDED_SECURITY set.",
-            "anonymous"
-          ])
-      ])
-
+        OptString.new('CAINPWFILE', [ false, 'Name of file to store Cain&Abel hashes in. Only supports NTLMv1 hashes. Can be a path.', nil ]),
+        OptString.new('JOHNPWFILE', [ false, 'Name of file to store JohnTheRipper hashes in. Supports NTLMv1 and NTLMv2 hashes, each of which is stored in separate files. Can also be a path.', nil ]),
+        OptString.new('CHALLENGE', [ false, 'The 8 byte server challenge. If unset or not a valid 16 character hexadecimal pattern, a random challenge is used instead.' ]),
```

Comment:
You may be able to use the `regex` kwarg to the `OptString` here to validate that the option is in the format that you're expecting.